holopy/core/metadata.py

```diff
@@ -233,3 +244,21 @@ def theta_phi(self, origin=None):
             for j, phi in enumerate(self.phi):
                 pos[i*self.shape[1]+j] = theta, phi
         return pos
+
+def interpret_args(schema=None, index=None, wavelen=None, polarization=None, optics=None):    
+    from .marray import Schema
+    if not isinstance(schema, Schema):
+        schema = Schema(positions=schema)
+    if optics is not None:
+        schema.optics=optics   
+    elif not isinstance(schema.optics, Optics):
+        schema.optics=Optics()
+
+    if index is not None:
+        schema.optics.index = index
+    if wavelen is not None:
+        schema.optics.wavelen = wavelen
+    if polarization is not None:
+        schema.optics.polarization = polarization
+    return schema
+
```

Comment:
I usually prefer making a new object and setting arguments in the constructor as opposed to the property setting you are doing here. Python passes things by reference, so if you have an existing optics on your schema, this code will change it, which could suprise you if you use the same schema in different places and only set the keywords in some of them.